### ISSUE TITLE
Restricted groups for communities

### ISSUE THREAD

**Issue by seanh** [hypothesis/product-backlog#426]
As a community I want to have a community controlled and branded group that anyone can read, but that only invited members can write to, so that I can limit annotations to my community members and not need to moderate the annotations.

**Note**: this is an **Epic** - a big user story that's too big to go into a sprint, it'll need to be fleshed out and split up into smaller stories before we can start working on it.

Notes:

* Same issue as for [open groups for communities](https://github.com/hypothesis/product-backlog/issues/425): when does the group appear in the groups dropdown menu for non-members?
* Same issues as with [restricted groups for publishers](https://github.com/hypothesis/product-backlog/issues/424).

**Comment by klemay** [comment]
Request from education user in [Zendesk ticket 2155](https://hypothesis.zendesk.com/agent/tickets/2155): 
> We had students contribute to our developing website explorecommonsense.com using the group feature of hypothes.is. We had the impression that one could control whether group annotations would display publicly or privately on the site. We have learned with much disappointment after students have already contributed their annotations that this is not the case. Is there any way we can have these annotations display publicly? We are very proud of their work and want to show it off.

**Comment by leonoverweel** [comment]
Any word on this? It'd be great if restricted groups could be used for this by anyone with an account on hypothes.is!

**Comment by leonoverweel** [comment]
This also seems to overlap quite a bit with #236.

**Comment by klemay** [comment]
@leonoverweel the feature that comes out of this work will go through a pilot phase with some initial partners, but our plan is to eventually make this available to all users! (And of course, the code that enables this feature will be available in our codebase.) 

This is separate from the work required for #236, which we haven't started planning yet, though #236 is also high priority for us!

**Comment by klemay** [comment]
Our current work on the Epic https://github.com/hypothesis/product-backlog/issues/866 will expand significantly on restricted groups for communities. When this work is done, the following will be true:

Members of restricted groups for communities will:  

- see their group in the groups scope selector across the web  
- be able to annotate in that group across the web
- share links to those annotations to members and non-members. Members and non-members will be able follow those links to view annotations in the group.

Out of scope for #866:
 - @seanh's question above— when does the group appear in the groups dropdown menu for non-members? When #866 is done, the group will only appear in the groups dropdown for non-members when they follow a share link created by a member.

**Comment by leonoverweel** [comment]
Hi @klemay, this sounds great! I have a question about this bullet:

> share links to those annotations to members and non-members. Members and non-members will be able follow those links to view annotations in the group.

Does this mean a member of a restricted group will be able to link to a specific, single annotation to share it with the outside world, or also (what I'd really like to see) link to something that shows all the annotations made by members of the restricted group on that page?

Thanks for the update!

**Comment by ajpeddakotla** [comment]
@leonoverweel short answer your questions, yes. 

With the work outlined in #866, #864, #855, you'll be able to share a direct link to a single annotation (something that is already possible), and share a link to the annotations made within a group on a specific page.

**Comment by leonoverweel** [comment]
Hi, just checking back on this. From #864, it looks like what I was looking for in my December 23rd comment has been implemented. I'm wondering now how (if?) I can make a group I have "world-readable" so that I can generate and share these kinds of links. On the edit page, I see options for renaming my group and changing its description, and I see a link I can use to invite people to contribute, but I don't see a toggle for making the group publicly readable. Am I looking in the wrong place? Thanks.

**Comment by klemay** [comment]
Hi @leonoverweel it's not possible to convert a private group into world-readable group (known at Hypothesis as a "restricted group"). You can use the direct linking feature with members of your private group, but would need a restricted group to share links to annotations with non-members.

Currently restricted groups are experimental and available to a few early partners. If you're interested in piloting this, shoot me an email at support@hypothes.is with some info about your use case.
